Bottom: 4d8da145950e7aa43dde18dfc165a3070e6ec01a
Top:    18979e972882b166a2398bdf7854d1d84f09ace2
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-01 17:10:53 +0000

Add the bloat as a parameter

Ok, this is nasty.. I'm obscuring setting of the global varaible... be careful!


---

diff --git a/src/find.c b/src/find.c
index c765bb1..2a20618 100644
--- a/src/find.c
+++ b/src/find.c
@@ -3326,8 +3326,10 @@ struct drc_info
 
 static void
 start_do_it_and_dump (int type, void *ptr1, void *ptr2, void *ptr3,
-                      int flag, bool AndDraw)
+                      int flag, bool AndDraw,
+                      Coord bloat)
 {
+  Bloat = bloat;
   ListStart (type, ptr1, ptr2, ptr3, flag);
   DoIt (flag, true, AndDraw);
   DumpList ();
@@ -3349,8 +3351,7 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
 
   if (PCB->Shrink != 0)
     {
-      Bloat = -PCB->Shrink;
-      start_do_it_and_dump (What, ptr1, ptr2, ptr3, DRCFLAG | SELECTEDFLAG, false);
+      start_do_it_and_dump (What, ptr1, ptr2, ptr3, DRCFLAG | SELECTEDFLAG, false, -PCB->Shrink);
       /* ok now the shrunk net has the SELECTEDFLAG set */
       ListStart (What, ptr1, ptr2, ptr3, FOUNDFLAG);
       Bloat = 0;
@@ -3362,11 +3363,9 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
           ClearFlagOnAllObjects (false, FOUNDFLAG | SELECTEDFLAG);
           User = true;
           drc = false;
-          Bloat = -PCB->Shrink;
-          start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, true);
-          Bloat = 0;
+          start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, true, -PCB->Shrink);
           drc = true;
-          start_do_it_and_dump (What, ptr1, ptr2, ptr3, FOUNDFLAG, true);
+          start_do_it_and_dump (What, ptr1, ptr2, ptr3, FOUNDFLAG, true, 0);
           User = false;
           drc = false;
           drcerr_count++;
@@ -3398,8 +3397,7 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
   /* now check the bloated condition */
   drc = false;
   ClearFlagOnAllObjects (false, FOUNDFLAG | SELECTEDFLAG);
-  Bloat = 0;
-  start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, false);
+  start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, false, 0);
   flag = FOUNDFLAG;
   ListStart (What, ptr1, ptr2, ptr3, flag);
   Bloat = PCB->Bloat;
@@ -3411,11 +3409,9 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
       ClearFlagOnAllObjects (false, FOUNDFLAG | SELECTEDFLAG);
       User = true;
       drc = false;
-      Bloat = 0;
-      start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, true);
-      Bloat = PCB->Bloat;
+      start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, true, 0);
       drc = true;
-      start_do_it_and_dump (What, ptr1, ptr2, ptr3, FOUNDFLAG, true);
+      start_do_it_and_dump (What, ptr1, ptr2, ptr3, FOUNDFLAG, true, PCB->Bloat);
       drcerr_count++;
       LocateError (&x, &y);
       BuildObjectList (&object_count, &object_id_list, &object_type_list);
@@ -3442,8 +3438,7 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
       Undo (true);
       /* highlight the rest of the encroaching net so it's not reported again */
       flag = FOUNDFLAG | SELECTEDFLAG;
-      Bloat = 0;
-      start_do_it_and_dump (thing_type, thing_ptr1, thing_ptr2, thing_ptr3, flag, true);
+      start_do_it_and_dump (thing_type, thing_ptr1, thing_ptr2, thing_ptr3, flag, true, 0);
       drc = true;
       Bloat = PCB->Bloat;
       ListStart (What, ptr1, ptr2, ptr3, flag);
